core/wallet/token.go

```diff
@@ -454,6 +454,20 @@ func (w *Wallet) TokensReceived(did string, ti []contract.TokenInfo, b *block.Bl
 	return nil
 }
 
+func (w *Wallet) TokenStateHashUpdate(tokenwithtokenhash []string) {
```

Comment:
Cant you write `TokenStateHash`  to the `tokenstable` the first time when you add token to the table ? Here you are writing to db twice. First write token details & then `TokenStateHash` separately to same row.